### DISCUSSION TITLE
BattleSnake Mono-thread

### DISCUSSION THREAD

**Discussion by naomi-lgbt** [EddieHubCommunity/support#896] [Conference Notes]
# Battlesnakes!

I've created this thread to serve as a place for you all to share your Battlesnake code, links, questions, and thoughts.

**Comment by eddiejaoude** [comment]
I am super excited to see what other people create

**Reply by eddiejaoude** [reply]
oh we should remind people about this

**Comment by Vyvy-vi** [comment]
Q) What strategy would you advice to get the snake's logic up fast?

**Reply by eddiejaoude** [reply]
I would recommend using one of their starter projects, this will move the snake in random directions, but it will work. Then I would suggestion

1. avoid colliding with the walls
2. avoid colliding with itself
3. avoid other snakes
4. chase food

**Reply by SashoStoichkov** [reply]
That sounds like a perfect plan 👍🏻

**Reply by Vyvy-vi** [reply]
hmm how's this for better snake-logic:
- shortest path to food
- longest path to it's own tail

**Reply by SashoStoichkov** [reply]
This sounds pretty similar to the `hamiltonian cycle` 👍🏻

**Reply by Vyvy-vi** [reply]
hmm however, the implementation might differ since the hamiltonian cycle is fixed for each match and the snake can short the cycle for food(I have no clue how to do the short-circuiting of the cycle for food and other snakes etc). I suppose these three can be integrated to make a great logic but I have an hour and my previous iterations were pretty much terrible. :joy:

**Comment by SashoStoichkov** [comment]
Here is my "python" - https://github.com/SashoStoichkov/BattleSnakePython100

**Comment by RashidKisome** [comment]
Anyone know a good starter repo for JS or TS repo(too lazy to look at the docs right now), I am gonna try and build one today.

**Reply by galahad42** [reply]
Here you'll find starter packs for all languages - https://docs.battlesnake.com/references/starter-projects
